src/http_request.cpp

```diff
@@ -114,8 +114,8 @@ const std::string http_request::get_arg(const std::string& key) const {
     return get_connection_value(key, MHD_GET_ARGUMENT_KIND);
 }
 
-const std::map<std::string, std::string, http::arg_comparator> http_request::get_args() const {
-    std::map<std::string, std::string, http::arg_comparator> arguments;
+const http::arg_map http_request::get_args() const {
```

Comment:
would it make sense for this to return a map of view as for the rest?

Reply:
Actually ran into an issue with this. `build_request_args` actually makes a temp copy of the underlying data and does some unescape stuff on it. We cannot return a view over that temp data since it goes out of scope.

Reply:
uhm...that is a really annoying inconsistency in the interface that will leave folks puzzled. I need to think of whether there is a way out (philosophically, I'd rather prioritize clarity/consistency over performance, unless major)

Reply:
To be fair, all of this originates from some old "unescaping" logic that made us bypass the mhd unescaper and build a separated unescaping logic on top of it. I think we might have to unblock that first (and I think it should be possible, given that nobody has touched that code in the past 10 years and existed to fix some misbehavior that was there in a dependency at that time).

Reply:
Agree that can be handled as a separate issue. Afterward we can make get_args return an arg_view_map.

Reply:
Do we think the MHD unescaper is fine at this point? Maybe worth checking.

Reply:
get_args firsts takes a copy of the args member var, then goes to MHD to get additional args. Can we use the args member var to just cache the unescaped values from MHD, and then return an arg_view_map constructed from the args member var?

Reply:
Sorry for the barrage of comments, but I notice get_arg to get a single arg does not do any unescaping, why is that?

Reply:
> Sorry for the barrage of comments, but I notice get_arg to get a single arg does not do any unescaping, why is that?

It sounds very much like a bug. To be fair, I can see there is some old code in here that makes no sense anymore (see it tries to get data from an "args" dataset that never gets filled because the method is const). I think the general management of args and unescaping has issues that need fixing.